populate-the-supabase/action.js

```diff
@@ -18,81 +42,223 @@ async function run() {
+      const parsedData = JSON.parse(content);
       const starsData = data.content.length > 0
 
-      // update data with repo id
-      for (const item of parsedData) {
-        const [owner, repo] = item.full_name.split("/")
-        const currentRepoResponse = await octokit.rest.repos.get({owner, repo})
-        const {
-          id,
-          stargazers_count,
-          description,
-          open_issues,
-        } = currentRepoResponse.data
-
-        const persistedData = await api.persistedRepoDataFetch({owner: owner, repo: repo})
-        const {contributors_oneGraph} = persistedData.data.gitHub.repositoryOwner.repository;
-
-        const contributorNames = await fetchContributorNames(contributors_oneGraph.nodes)
-
-        item.id = id
-
-        await supabase.from('user_stars').insert({
-          id: repository.id,
-          user_id: repository.owner.id,
-          star_id: item.id,
-          repo_name: item.full_name,
-          recency_score: parsedData.indexOf(item),
-          description: description,
-          issues: open_issues,
-          stars: stargazers_count,
-          contributors: contributorNames.slice(0,2) // grab first two names only
-        })
-      }
+      parseData.push({
+        installation,
+        parsedData,
+        starsData,
+        repository,
+      })
+    }
+  }
 
-      // send parsedData to stars table
-      await supabase.from('stars').upsert(parsedData, {onConflict: "id"})
+  consoleHeader('Parsing stars');
+  console.log(`Existing installation queue was ${parseInstallations.length}`)
+  console.log(`Attempting to parse ${parseData.length} users out of ${limitUsers} process.env.LIMIT_USERS`)
 
-      console.log(`ADDED STARS FROM: ${repository.html_url}`)
+  await Promise.all(
+    parseData.map(async ({installation, parsedData, starsData, repository}) =>
+      new Promise(async (resolve, reject) => {
+        try {
+          console.log(`Processing installation #${installation.id}, ${installation.account.login} stars.json`)
+          const octokit = await app.getInstallationOctokit(installation.id);
 
-      // send parsedData to supabase
-     await supabase.from('users')
-      .upsert({id: repository.owner.id, stars_data: starsData,
-        open_issues: repository.open_issues, private: repository.private}, {
-          onConflict:
-          'id'
-        })
+          // update data with repo id
+          for await (const item of parsedData) {
+            const [owner, repo] = item.full_name.split("/")
+            const currentRepoResponse = await octokit.rest.repos.get({owner, repo})
+              .catch((err) => {
+                return {errors: [err]}
+              })
 
-      console.log(`ADDED USER FROM: ${repository.owner.login}`)
+            if (currentRepoResponse.errors && currentRepoResponse.errors.length > 0) {
+              console.log(`ERROR for ${owner}/${repo}`, currentRepoResponse.errors)
+              // reject(currentRepoResponse.errors)
```

Comment:
I have not used this reject syntax before, but is this meant to be commented out?

Reply:
it was initially throwing everywhere but in some cases those errors weren't crucial so ended up just logging them - it's not the best looking piece of code but it does what it's supposed to 🍕